pallets/committee/src/lib.rs

```diff
@@ -380,33 +393,41 @@ pub mod pallet {
 		pub fn close(origin: OriginFor<T>, proposal_hash: HashFor<T>) -> DispatchResultWithPostInfo {
 			let closer = T::ProposalExecutionOrigin::ensure_origin(origin)?;
 
-			// ensure proposal has not already been executed
-			ensure!(!ExecutedProposals::<T>::contains_key(proposal_hash), Error::<T>::ProposalAlreadyExecuted);
+			// register that this proposal has been executed
+			Proposals::<T>::try_mutate_exists(&proposal_hash, |maybe_proposal| -> DispatchResultWithPostInfo {
+				let mut proposal = maybe_proposal.take().ok_or(Error::<T>::NoProposalWithHash)?;
 
-			let votes = Self::get_votes_for(&proposal_hash).ok_or(Error::<T>::NoProposalWithHash)?;
-			let current_block = frame_system::Pallet::<T>::block_number();
+				// ensure proposal has not already been executed
+				(match proposal.status {
+					ProposalStatus::Closed => Err(Error::<T>::ProposalAlreadyClosed),
+					ProposalStatus::Executed => Err(Error::<T>::ProposalAlreadyExecuted),
+					_ => Ok(()),
+				})?;
 
-			// Ensure voting period is over
-			ensure!(current_block > votes.end, Error::<T>::VotingPeriodNotElapsed);
+				let votes = Self::get_votes_for(&proposal_hash).ok_or(Error::<T>::NoProposalWithHash)?;
+				let current_block = frame_system::Pallet::<T>::block_number();
 
-			// Ensure voting has accepted proposal
-			votes.is_accepted(T::MinCouncilVotes::get()).map_err(Into::<Error<T>>::into)?;
+				// Ensure voting period is over
+				ensure!(current_block > votes.end, Error::<T>::VotingPeriodNotElapsed);
 
-			// Execute the proposal
-			let proposal = Self::get_proposal(&proposal_hash).ok_or(Error::<T>::NoProposalWithHash)?;
-			let result = proposal.1.dispatch(Origin::<T>::ApprovedByCommittee(closer, votes).into());
```

Comment:
shouldn't this be the original issuer of the proposal?

Reply:
https://github.com/ChainSafe/PINT/issues/373#issuecomment-919839363 should be another pr since this requires to fix the tests and benchmarks together